src/compiler/scala/tools/nsc/backend/jvm/opt/LocalOpt.scala

```diff
@@ -612,16 +613,21 @@ object LocalOptImpls {
       })
     }
 
-    var removedHandlers = Set.empty[TryCatchBlockNode]
+    var handlerRemoved = false
+    var liveHandlerRemoved = false
+
     val handlersIter = method.tryCatchBlocks.iterator()
     while (handlersIter.hasNext) {
       val handler = handlersIter.next()
       if (!containsExecutableCode(handler.start, handler.end)) {
-        removedHandlers += handler
+        handlerRemoved = true
+        if (!liveHandlerRemoved && BackendUtils.isLabelReachable(handler.start))
+          liveHandlerRemoved = true
         handlersIter.remove()
       }
     }
-    removedHandlers
+
+    (handlerRemoved, liveHandlerRemoved)
```

Comment:
The benchmarks suggest this commit made performance go slightly backwards.

Assuming the common case is that nothing is removed, it could help to reuse an instance of `(false, false)` here (and above on line 323), which would be allocation free.

Reply:
The code will probably look a bit cleaner if you introduce a case class to represent that result.

```
case class RemoveEmptyExceptionHandlersResult(handlerRemoved: Boolean, liveHandlerRemoved: Boolean)
object RemoveEmptyExceptionHandlersResult {
   val Empty = RemoveEmptyExceptionHandlersResult(false, false)
   def get(handlerRemoved: Boolean, liveHandlerRemoved: Boolean) = if (handlerRemoved || liveHandlerRemoved) RemoveEmptyExceptionHandlersResult(handlerRemoved, liveHandlerRemoved) else Empty
}
```

Reply:
I can also encode the two booleans in an int

Reply:
done